idp-binding-service/src/main/java/io/mosip/idp/binding/repository/PublicKeyRegistryRepository.java

```diff
@@ -25,15 +25,17 @@ public interface PublicKeyRegistryRepository extends JpaRepository<PublicKeyRegi
 	 * @param idHash
 	 * @return
 	 */
-	Optional<PublicKeyRegistry> findByIdHash(String idHash);
+	Optional<PublicKeyRegistry> findByIdHashAndExpiredtimesGreaterThan(String idHash, LocalDateTime currentDateTime);
 
-    @Query("SELECT pkr FROM PublicKeyRegistry pkr WHERE pkr.psuToken= :psuToken")
+	@Query(value = "SELECT * FROM public_key_registry WHERE psu_token= :psuToken ORDER BY expire_dtimes DESC LIMIT 1", nativeQuery = true)
 	Optional<PublicKeyRegistry> findOneByPsuToken(String psuToken);
```

Comment:
Rename this method to findLatestByPSUToken

Reply:
Done